service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/EhrExtractMapper.java

```diff
@@ -67,13 +57,17 @@ private EhrExtractTemplateParameters prepareEhrFhirExtractParamsFromFhirBundle(
     }
 
     private Optional<Patient> extractPatientFromBundle(Bundle bundle) {
-        return bundle.getEntry()
-            .stream()
-            .filter(entry -> !entry.isEmpty())
-            .map(Bundle.BundleEntryComponent::getResource)
-            .filter(resource -> resource.getResourceType() == ResourceType.Patient)
-            .map(resource -> (Patient) resource)
-            .findFirst();
+        if (bundle != null && bundle.hasEntry()) {
```

Comment:
I'm confused as to why this method exists at all. The returned patient is never used anywhere, we just throw an exception if its missing.

Reply:
The verification of patient in the bundle was removed due to misunderstanding with creating it at the beginning.